junebug/tests/helpers.py

```diff
@@ -1,13 +1,48 @@
 from copy import deepcopy
 import logging
+from twisted.internet import reactor
 from twisted.internet.defer import inlineCallbacks, returnValue
 from twisted.trial.unittest import TestCase
+from twisted.web.server import Site
+from txamqp.client import TwistedDelegate
+from vumi.utils import vumi_resource_path
+from vumi.service import get_spec
+from vumi.tests.fake_amqp import FakeAMQPBroker, FakeAMQPChannel
 from vumi.tests.helpers import PersistenceHelper, WorkerHelper
 
+from junebug import JunebugApi
+from junebug.amqp import JunebugAMQClient
 from junebug.channel import Channel
 from junebug.service import JunebugService
 
 
+class FakeAmqpFactory(object):
+    def __init__(self, amqp_client):
+        self.amqp_client = amqp_client
+
+    def get_client(self):
+        return self.amqp_client
+
+
+class FakeAmqpClient(JunebugAMQClient):
+    def __init__(self, spec):
+        super(FakeAmqpClient, self).__init__(TwistedDelegate(), '', spec)
+        self.broker = FakeAMQPBroker()
+
+    @inlineCallbacks
+    def channel(self, id):
+        yield self.channelLock.acquire()
+        try:
+            try:
+                ch = self.channels[id]
+            except KeyError:
+                ch = FakeAMQPChannel(id, self)
+                self.channels[id] = ch
+        finally:
+            self.channelLock.release()
```

Comment:
I see. I think we should maybe try avoid duplicating stuff vumi is doing. One way could be to have our `FakeAmqpClient` mixin `JunebugAMQClient` and vumi's `FakeAMQClient`.

Alternatively, maybe if we're doing mixins, its a sign that we need to using composition instead of inheritance (so have `JunebugAMQClient` have the amqp client as an instance variable passed to it via `__init__` or something).

Thoughts?